main.c

```diff
@@ -1,167 +1,56 @@
+            case 3: {// Choose the drink
                 printf("Choose your drink:\n");
-                for(int i=0;i<nodrinks;i++) {
-                    putchar('a'+i);
-                    printf(") %s (%.2f)\n", drinks[i], drinksPrices[i]);
-                }
-                printf("%c) Go back\n",'a'+nodrinks);
-
-                //we want to check here for '\n' to allow the user to select 0 additional items
-                drinkChoice = getchar();
-                getchar(); //consume new line
-                if(drinkChoice == 'a'+nodrinks) {
-                    state--;
-                    break;
-                }
-                    drinkChoice = drinkChoice-'a';
-                state++;
-                break;
-            }
-            case 4:
-            {
-                //cutlery
-                printf("Do you want cutlery?\n");
-                printf("a) Yes\nb) No\nc) Go back\n");
-                cutlery=getchar();
-                getchar(); //consume new line
-                if (cutlery=='c') state--;
-                else state++;
-                break;
-            }
-            case 5:
-            {
-                //Additional Info
-                printf("Any additional info? Type 'no' for no message.\n");
-                scanf("%s",  infoMsg);
-                if (infoMsg[0]=='n' && infoMsg[1]=='o' && infoMsg[2]==NULL) info=1;
-                else info=0;
-                state++;
-            }
-            case 6:{
-                // Display contract
-                printf("Your order is:\n");
-                printf("-------------\n");
-                printf("-username: %s\n", username);
-                printf("Food: %s (%.2f)\n", types[mealChoice][typeChoice], prices[mealChoice][typeChoice]);
-                printf("Drinks: %s (%.2f)\n", drinks[drinkChoice], drinksPrices[drinkChoice]);
-                printf("Cutlery: ");
-                printf(cutlery ? "No\n" : "Yes\n");
-                if (!info) printf("Note: %s\n", infoMsg);
+                showFoodWithPrice(nodrinks, drinks, drinksPrices);
+                drinkChoice=getchar();  getchar();
+                checkBackOption(drinkChoice, nodrinks, &state);
+                drinkChoice = drinkChoice - 'a'; state++; break;}
+            case 4: {//cutlery
+                askCutlery();
+                cutlery=getchar()-'a';  getchar();
+                checkBackOption(cutlery+'a', 'c'-'a', &state);
+                state++; break;}
+            case 5: {printf("Any additional info?\n");
+                gets(infoMsg); state++;
+                if (strcmp(infoMsg, "\n") == 0) info = 0;   else info = 1;}
+            case 6: {// Display contract
+                printUser(username);
+                printOrder(item1, types[mealChoice][typeChoice], prices[mealChoice][typeChoice]);   printOrder(item2, drinks[drinkChoice], drinksPrices[drinkChoice]);
+                printCutleryAndMessage(cutlery, info, infoMsg);
```

Comment:
and is a hint for splitting functionaity - one function should do one thing and one thing only